### ISSUE TITLE
Querying a Draft containing a Neo field with nested blocks (level 4) using Craft's Graphiql playground returns an empty result when modified

### ISSUE THREAD

**Issue by nedu64** [spicywebau/craft-neo#908]
### Bug Description

I currently have a  **Neo** field which contains a few blocks used in adding various content to a page. Based on the Block configuration and the issue at hand, i noticed adding child blocks to an existing child block up to level 4 seems a bit problematic when querying its modified Drafts content as tested using Graphiql playground. Here is a bit of description of the block configuration:

- Field name:Content Builder
- Field type: Neo
- Block Configuration
   - Content Row
      - Content Column (Child Block)
         - Accordion Item (Child Block)
            - Heading 
            - Hidden Content (Child Block)
               - Buttons (Child Block)
               - Copy (Child Block)
               - Heading (Child Block)
               - Downloads Dropdown Buttons (Child Block)
               - Downloads Dropdown Block (Child Block)
               - ...

**Pictorial Description**

<img width="971" alt="Screenshot 2024-07-08 at 10 30 15" src="https://github.com/spicywebau/craft-neo/assets/173365706/662f56b2-4ffb-410b-baf4-3871a6ff1e01">
<img width="886" alt="Screenshot 2024-07-08 at 10 29 21" src="https://github.com/spicywebau/craft-neo/assets/173365706/6037b959-e1f5-4c90-bdf1-527733bf78f7">
<img width="1566" alt="Screenshot 2024-07-08 at 09 54 01" src="https://github.com/spicywebau/craft-neo/assets/173365706/4b972ec9-4315-49c9-9f3d-208d93d1a2bf">
<img width="876" alt="Screenshot 2024-07-08 at 10 34 49" src="https://github.com/spicywebau/craft-neo/assets/173365706/bedf1dad-7963-4cf6-a4be-e4e07201b06e">

Adding content to any of the **Hidden Content's** Block works in Live Preview when saved. However if you create a Draft from the saved page and make any modifications to any of the **Hidden Content's** Child Blocks, Graphiql playground returns an empty response for the Child Blocks:
E.g:

Saved content below

<img width="1140" alt="Screenshot 2024-07-08 at 10 19 19" src="https://github.com/spicywebau/craft-neo/assets/173365706/81621ccf-f463-450c-8267-c4b43e8ecadb">

Query (Graphiql):
```javascript
contentBuilder {
          ... on contentBuilder_contentRow_BlockType {
            children {
              ... on contentBuilder_contentColumn_BlockType {
                children {
                  ... on contentBuilder_accordion_BlockType {
                    heading
                    children {
                      ... on contentBuilder_richText_BlockType {
                        __typename
                        richText
                      }
                      ... on contentBuilder_pane_BlockType {
                        id
                        children {
                          ... on contentBuilder_richText_BlockType {
                            id
                            richText
                          }
                        }
                      }
                    }
                  }
                  ... on contentBuilder_richText_BlockType {
                    __typename
                    richText
                  }
                }
              }
            }
          }
        }
``` 

Response:
```javascript
 {
      "heading": "Test heading",
      "children": [
        {
          "id": "1429731",
          "children": [
            {
              "id": "1786121",
              "richText": "<p>Saved content currently works</p>"
            }
          ]
        }
      ]
    },
``` 
- Create a draft from the saved page
   - Add changes to the `Accordion item` block e.g:  
<img width="1061" alt="Screenshot 2024-07-08 at 11 02 07" src="https://github.com/spicywebau/craft-neo/assets/173365706/195c944c-a4f1-4325-8d89-e1585743991e">
    - Query the saved draft in graphiql playground
       - Query
       
```javascript
drafts {
      id
      title
      ... on pages_builder_Entry {
        contentBuilder {
          ... on contentBuilder_contentRow_BlockType {
            children {
              ... on contentBuilder_contentColumn_BlockType {
                children {
                  ... on contentBuilder_accordion_BlockType {
                    heading
                    children {
                      ... on contentBuilder_richText_BlockType {
                        __typename
                        richText
                      }
                      ... on contentBuilder_pane_BlockType {
                        id
                        children {
                          ... on contentBuilder_richText_BlockType {
                            id
                            richText
                          }
                        }
                      }
                    }
                  }
                  ... on contentBuilder_richText_BlockType {
                    __typename
                    richText
                  }
                }
              }
            }
          }
        }
      }
    }
```  
Response:
```javascript
{
  "heading": "Test heading",
  "children": []
},
``` 

### Steps to reproduce

1. Create a test page in craft
2. Add a Neo field
3. Model your Neo field based on the example shared in the **Bug Description** section
    - Field name: Content Builder
    - Field type: Neo
    - Block Configuration
       - Content Row
          - Content Column (Child Block)
             - Accordion Item (Child Block)
                - Heading 
                - Hidden Content (Child Block)
                    - Buttons (Child Block)
                    - Copy (Child Block)
                    - Heading (Child Block)
                    - Downloads Dropdown Buttons (Child Block)
                    - Downloads Dropdown Block (Child Block)
4. Add contents to the (Level 4) nested block (Hidden Content)
5. Confirm that you can query your saved content using **Graphiql Playground**
6. Create a draft from your saved page containing modelled Neo field
7. Add modifications to the (Level 4) nested block (Hidden Content)
8. Try querying Graphiql Playground again for the changes added to your saved drafts
9. Note that all child blocks of the **Hidden Content** (Level 4) Block are missing

### Expected behaviour

Adding changes to a nested block up to Level 4 in a created Draft, i'm able to see these changes when querying using Craft's Graphiql Playground

### Neo version

4.1.2

### Craft CMS version

Craft Pro 4.9.5

### Does this issue involve templating, and if so, is eager-loading used?

This is not a templating issue

**Comment by johnnynotsolucky** [comment]
I think this may be loosely related to the issue reported in #899 as well.

**Comment by ttempleton** [comment]
If possible, could you please send your composer.json/lock files and database backup to plugins@spicyweb.com.au, along with directions to an entry where this reliably happens, and we'll have a look.

**Comment by johnnynotsolucky** [comment]
Thanks for looking into this @ttempleton. We won't be able to send a copy of the database for the install that we're seeing this issue on. However, we're going to test with version 4.2.7 of Neo, and if the issue we're seeing is still present, we'll set up a fresh Craft install and try and reproduce it - we would then be able to send you a copy of that DB.

**Comment by nedu64** [comment]
> If possible, could you please send your composer.json/lock files and database backup to [plugins@spicyweb.com.au](mailto:plugins@spicyweb.com.au), along with directions to an entry where this reliably happens, and we'll have a look.

@ttempleton  Done, also note that the issue is also reproducible  in version `4.2.8`

**Comment by ttempleton** [comment]
Thanks for that @nedu64 - fixed now in 4.2.9.
